TaskfMRIAnalysis/TaskfMRIAnalysis.sh

```diff
@@ -39,56 +39,92 @@
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
 
-${script_name}: Run TaskfMRIAnalysis pipeline
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/fsl_version.shlib"          # Function for getting FSL version
 
-Usage: ${script_name} [options]
 
-Usage information To Be Written
+# --------------------------------------------------------------------------------
+#  Usage Description Function
+# --------------------------------------------------------------------------------
 
-EOF
+#this function gets called by opts_ParseArguments when --help is specified
+function usage()
+{
+    #header text
+    echo "
+$log_ToolName: Run TaskfMRIAnalysis pipeline for a subject. Pipeline will run Level1 (scan-level) analyses, and Level2 (single subject-level) analysis as specified.
+
+Usage: $log_ToolName arguments...
+[ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
 }
 
-# Allow script to return a Usage statement, before any other output or checking
-if [ "$#" = "0" ]; then
-    show_usage
-    exit 1
-fi
-
 # ------------------------------------------------------------------------------
-#  Check that HCPPIPEDIR is defined and Load Function Libraries
+#  Parse Command Line Options
 # ------------------------------------------------------------------------------
 
-if [ -z "${HCPPIPEDIR}" ]; then
-  echo "${script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
-  exit 1
-fi
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+opts_AddMandatory '--study-folder' 'Path' '/path/to/study/folder' "directory containing imaging data for all subjects"
+opts_AddMandatory '--subject' 'Subject' 'SubjectID' ""
+opts_AddMandatory '--lvl1tasks' 'LevelOnefMRINames' 'tfMRI_TaskName_RL@tfMRI_TaskName_LR' "List of subdirectories containing each time series image for the TaskName task. Multiple scan subdirectories should be provided as a single string separated by '@' character." #Assumes these subdirectories are located in the SubjectID/MNINonLinear/Results directory. Also assumes that timeseries image filename begins with this string.
+opts_AddOptional '--lvl1fsfs' 'LevelOnefsfNames' 'DesignName_TaskName_RL@DesignName_TaskName_LR' "List of prefixes in the design.fsf filenames for each time series image for the TaskName task. Should contain same number of design files as time series images in --lvl1tasks option. (N-th design will be used for N-th time series image.) Separate multiple design files by '@' character. If no value is passed to --lvl1fsfs, the value will be set to the same list passed to --lvl1tasks."
```

Comment:
"List of fsf file names (don't include .fsf extension) for each fMRI scan name..."

Reply:
Similar to above, the script is specifically expecting the prefix of the design file naming convention. The user should not enter "prefix_hp200_s4". The other portions of the file name are currently determined by other variables.

Reply:
Where does the user learn that the `.fsf` are supposed to have `hpX_sY` as part of their file name?

Reply:
Changed example to 'DesignName1@DesignName2'.

The file naming conventions exist. If we don't show them in the arguments, I guess the user will need to learn the file naming conventions and CinaB directory structure on by familiarizing themselves with the data and the code.